core/src/main/java/com/datastax/oss/driver/internal/core/os/GraalGetpid.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.internal.core.os;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import org.graalvm.nativeimage.c.CContext;
+import org.graalvm.nativeimage.c.function.CFunction;
+
+@CContext(GraalGetpid.Directives.class)
+public class GraalGetpid {
+
+  static class Directives implements CContext.Directives {
+
+    @Override
+    public List<String> getHeaderFiles() {
+
+      return Collections.unmodifiableList(Arrays.asList("<unistd.h>"));
+    }
+
+    /* TODO: I don't know that these are required... */
+    @Override
+    public List<String> getMacroDefinitions() {
+
+      return Collections.unmodifiableList(Arrays.asList("_GNU_SOURCE", "_LARGEFILE64_SOURCE"));
```

Comment:
Isn't the shaded guava already included in the build via java-driver dependency?

Reply:
It is, but we're just creating a simple unmodifiable list here, either approach will do.

Reply:
It's not a question of whether the classes are available @tomekl007 ... you're correct, they certainly are in the classpath.  The point is that this code will be eval'd at build time (since we have to build C source off of it).  And if we include the Guava classes here you'll get errors in the native build operation saying that some Guava classes were unexpectedly discovered at build time, which you then have to work around by specifying the --initialize-at-build-time flag for each of those classes.  I'm trying to avoid all of that by keeping Guava out of this section of the code base.